code/game/objects/items/devices/scanners/breath.dm

```diff
@@ -0,0 +1,118 @@
+		header += "<style> .scan_danger{color: #ff0000; font-weight: bold;}</style>"
+		header += "<style> .scan_red{color:red}</style>"
+		header += "<style> .scan_green{color:green}</style>"
+		header += "<style> .scan_blue{color: #5f94af}</style>"
+		header += "<style> .scan_orange{color:#ffa500}</style>"
+		b		= "<b>"
+		endb	= "</b>"
+
+	. += "[b]Breath sample results for \the [C]:[endb]"
+
+	var/obj/item/organ/internal/lungs/lungs = C.get_organ(BP_LUNGS)
+	var/breathing = "none"
+	if(istype(lungs) && !(C.status_flags & FAKEDEATH))
+		if(lungs.breath_fail_ratio < 0.3)
+			breathing = "normal"
+		else if(lungs.breath_fail_ratio < 1)
+			breathing = "shallow"
+
+	switch(breathing)
+		if("none")
+			. += "<span class='scan_danger'>Alert: No breathing detected.</span>"
+			return
+		if("shallow")
+			. += "<span class='scan_warning'>Subject's breathing is abnormally shallow.</span>"
+		if("normal")
+			. += "<span class='scan_notice'>Subject's breathing is normal.</span>"
+
+	// Other general warnings.
+	if(skill_level >= SKILL_BASIC)
+		switch(C.getOxyLoss())
+			if(0 to 25)
+				dat += "<span class='scan_green'>Subject oxygen levels nominal.</span>"
+			if(25 to 50)
+				dat += "<span class='scan_orange'>Subject oxygen levels abnormal.</span>"
+			if(50 to INFINITY)
+				dat += "<span class='scan_blue'>[b]Severe oxygen deprivation detected.[endb]</span>"
+
+		if(istype(lungs))
+			if(lungs.is_broken())
+				dat += "<span class='scan_danger'>[b]Tension pneumothorax detected.[endb]</span>"
+			else if(lungs.is_bruised())
+				dat += "<span class='scan_warning'>Collapsed lung detected.</span>"
+			else
+				dat += "<span class='scan_green'>Subject lung health nominal.</span>"
+		else
+			dat += "<span class='scan_warning'>Subject lung health unknown.</span>"
+
+	. += (skill_level < SKILL_BASIC) ? shuffle(dat) : dat
+
+	// Reagent data.
+	. += "[b]Reagent scan:[endb]"
+
+	var/print_reagent_default_message = TRUE
+	if (C.has_chemical_effect(CE_ALCOHOL, 1))
+		. += "<span class='scan_orange'>Alcohol detected in subject's breath.</span>"
+		print_reagent_default_message = FALSE
+	if (C.has_chemical_effect(CE_ALCOHOL_TOXIC, 1))
+		. += "<span class='scan_red'>Subject is suffering from alcohol poisoning.</span>"
+		print_reagent_default_message = FALSE
+
+	var/datum/reagents/inhaled = C.get_inhaled_reagents()
+	if(inhaled && inhaled.total_volume)
+		var/unknown = 0
+		for(var/rtype in inhaled.reagent_volumes)
+			var/decl/material/R = GET_DECL(rtype)
+			if(R.scannable)
+				print_reagent_default_message = FALSE
+				. += "<span class='scan_notice'>[capitalize(R.gas_name)] found in subject's breath.</span>"
+			else
+				++unknown
+		if(unknown)
+			print_reagent_default_message = FALSE
+			. += "<span class='scan_warning'>Non-medical reagent[(unknown > 1)?"s":""] found in subject's breath.</span>"
+
+	if(print_reagent_default_message)
+		. += "No results."
+
+	header = jointext(header, null)
+	. = jointext(.,"<br>")
+	. = jointext(list(header,.),null)
```

Comment:
For what it's worth string operations were optimized in some DM update, this kind of stuff is no longer really needed.